src/energyplus/ForwardTranslator/ForwardTranslateUnitarySystemPerformanceMultispeed.cpp

```diff
@@ -34,44 +54,78 @@ namespace openstudio {
 namespace energyplus {
 
   boost::optional<IdfObject> ForwardTranslator::translateUnitarySystemPerformanceMultispeed(UnitarySystemPerformanceMultispeed& modelObject) {
-    boost::optional<std::string> s;
-    boost::optional<double> d;
 
     // Name
     IdfObject sysPerf = createRegisterAndNameIdfObject(openstudio::IddObjectType::UnitarySystemPerformance_Multispeed, modelObject);
 
-    // Find the associated AirLoopHVACUnitarySystem
+    boost::optional<HVACComponent> heatingCoil;
+    boost::optional<HVACComponent> coolingCoil;
+
+    // Find the associated AirLoopHVACUnitarySystem, ZoneHVACTerminalUnitVariableRefrigerantFlow, or ZoneHVACWaterToAirHeatPump
     std::vector<AirLoopHVACUnitarySystem> unitarySystems = modelObject.getModelObjectSources<AirLoopHVACUnitarySystem>();
-    if (unitarySystems.size() != 1) {
-      LOG(Error, "OS:UnitarySystemPerformance:Multispeed should be referenced by one and only one OS:AirLoopHVAC:UnitarySystem, "
-                   << modelObject.nameString() << " is referenced by " << unitarySystems.size() << ".")
+    std::vector<ZoneHVACTerminalUnitVariableRefrigerantFlow> tuVRFs =
+      modelObject.getModelObjectSources<ZoneHVACTerminalUnitVariableRefrigerantFlow>();
+    std::vector<ZoneHVACWaterToAirHeatPump> hps = modelObject.getModelObjectSources<ZoneHVACWaterToAirHeatPump>();
+
+    if (((unitarySystems.size() + tuVRFs.size() + hps.size()) != 1)) {
+      LOG(Error, "OS:UnitarySystemPerformance:Multispeed should be referenced by one and only one OS:AirLoopHVAC:UnitarySystem or "
+                 "OS:ZoneHVAC:TerminalUnit:VariableRefrigerantFlow or OS:ZoneHVAC:WaterToAirHeatPump, "
+                   << modelObject.nameString() << " is referenced by " << unitarySystems.size() << " OS:AirLoopHVAC:UnitarySystem, " << tuVRFs.size()
+                   << " OS:ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, " << hps.size() << " OS:ZoneHVAC:WaterToAirHeatPump.")
       return boost::none;
     }
-    AirLoopHVACUnitarySystem& unitarySystem = unitarySystems[0];
+
+    if (unitarySystems.size() == 1) {
+      AirLoopHVACUnitarySystem& unitarySystem = unitarySystems[0];
+      heatingCoil = unitarySystem.heatingCoil();
+      coolingCoil = unitarySystem.coolingCoil();
+    }
+
+    if (tuVRFs.size() == 1) {
+      ZoneHVACTerminalUnitVariableRefrigerantFlow& tuVRF = tuVRFs[0];
+      heatingCoil = tuVRF.heatingCoil();
+      coolingCoil = tuVRF.coolingCoil();
+    }
+
+    if (hps.size() == 1) {
+      ZoneHVACWaterToAirHeatPump& hp = hps[0];
+      heatingCoil = hp.heatingCoil().optionalCast<HVACComponent>();
+      coolingCoil = hp.coolingCoil().optionalCast<HVACComponent>();
+    }
```

Comment:
```suggestion
    if (unitarySystems.size() == 1) {
      AirLoopHVACUnitarySystem& unitarySystem = unitarySystems.front();
      heatingCoil = unitarySystem.heatingCoil();
      coolingCoil = unitarySystem.coolingCoil();
    } else if (tuVRFs.size() == 1) {
      ZoneHVACTerminalUnitVariableRefrigerantFlow& tuVRF = tuVRFs.front();
      heatingCoil = tuVRF.heatingCoil();
      coolingCoil = tuVRF.coolingCoil();
    } else {  // if (hps.size() == 1)
      ZoneHVACWaterToAirHeatPump& hp = hps.front();
      heatingCoil = hp.heatingCoil().optionalCast<HVACComponent>();
      coolingCoil = hp.coolingCoil().optionalCast<HVACComponent>();
    }
```

Reply:
Because of the `if (((unitarySystems.size() + tuVRFs.size() + hps.size()) != 1)) {` line above, yes, I think we'd be able to collapse to your suggestion.